### PR TITLE
[2.5] Add the hello-pt-resnet example

### PR DESCRIPTION
**Pull request by yhwen**
Fixes # .

### Description

Add an example to show how to wrap up the Python function into a Python class for creating an object instance to use in the job API. It uses the torchvision ResNet to replace the SImpleNetwork in the hello-pt example.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

**Review by yanchengnv — APPROVED** [review]
LGTM

**Comment by YuanTingHsieh** [comment]
same feedback as in #2954

File: examples/hello-world/hello-pt-resnet/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+Example of using [NVIDIA FLARE](https://nvflare.readthedocs.io/en/main/index.html) to train an image classifier
+using federated averaging ([FedAvg](https://arxiv.org/abs/1602.05629))
+and [PyTorch](https://pytorch.org/) as the deep learning training framework. Comparing with the Hello PyTorch example, it uses the torchvision ResNet, 
+instead of the SimpleNetwork.
+
+> **_NOTE:_** This example uses the [CIFAR-10](https://www.cs.toronto.edu/~kriz/cifar.html) dataset and will load its data within the client train code.
+
+The Job API only supports the object instance created directly out of the Python Class. It does not support 
+the object instance created through using the Python function. Comparing with the hello-pt example, 
+if we replace the SimpleNetwork() object with the resnet18(num_classes=10),
```
Comment (YuanTingHsieh) [inline examples/hello-world/hello-pt-resnet/README.md:12]: Suggest Add a link reference to this resnet18 method

File: examples/hello-world/hello-pt-resnet/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Hello PyTorch ResNet
```
Comment (YuanTingHsieh) [inline examples/hello-world/hello-pt-resnet/README.md:1]: Should this be a hello-xxx example?

File: examples/hello-world/hello-pt-resnet/src/hello-pt_cifar10_fl.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
```
Comment (YuanTingHsieh) [inline examples/hello-world/hello-pt-resnet/src/hello-pt_cifar10_fl.py:1]: 2024

File: examples/hello-world/hello-pt-resnet/src/simple_network.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
```
Comment (YuanTingHsieh) [inline examples/hello-world/hello-pt-resnet/src/simple_network.py:1]: Should we rename the file to be resnet.py?

**Review by YuanTingHsieh — COMMENTED** [review]
Content mostly good, some suggestions

File: examples/hello-world/hello-pt-resnet/README.md (hunk 1 above)
Comment (yhwen) [reply examples/hello-world/hello-pt-resnet/README.md:12]: Added link.

File: examples/hello-world/hello-pt-resnet/README.md (hunk 2 above)
Comment (yhwen) [reply examples/hello-world/hello-pt-resnet/README.md:1]: keep is as hello-pt-resnet for now. The implementation approach is the same.

File: examples/hello-world/hello-pt-resnet/src/simple_network.py (hunk 4 above)
Comment (yhwen) [reply examples/hello-world/hello-pt-resnet/src/simple_network.py]: renamed

**Review by YuanTingHsieh — APPROVED** [review]

**Pull request merged**